src/cairo_run.rs

```diff
@@ -49,16 +49,24 @@ pub fn write_output(cairo_runner: &mut CairoRunner) -> Result<(), CairoRunError>
 
 /// Writes a trace as a binary file. Bincode encodes to little endian by default and each trace
 /// entry is composed of 3 usize values that are padded to always reach 64 bit size.
-pub fn write_binary_trace(relocated_trace: &[RelocatedTraceEntry], trace_file: &Path) {
-    let mut buffer = File::create(trace_file).expect("Error while creating trace file");
+pub fn write_binary_trace(
+    relocated_trace: &[RelocatedTraceEntry],
+    trace_file: &Path,
+) -> io::Result<()> {
+    let mut buffer = File::create(trace_file).expect("Could not create trace file");
+
     for (i, entry) in relocated_trace.iter().enumerate() {
-        if let Err(e) = bincode::serialize_into(&mut buffer, entry) {
-            println!(
-                "Failed to dump trace at position {}, serialize error: {}",
-                i, e
-            );
+        match bincode::serialize_into(&mut buffer, entry) {
+            Ok(()) => continue,
+            Err(e) => {
+                let error_string =
+                    format!("Failed to dump trace at position {i}, serialize error: {e}");
+                return Err(Error::new(ErrorKind::Other, error_string));
+            }
```

Comment:
This looks like `if let Err(e)` still did the job, just changing the body to the new one.